src/System.CommandLine/Invocation/TypoCorrection.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace System.CommandLine.Invocation
+{
+    internal class TypoCorrection
+    {
+        private readonly int _maxLevenshteinDistance;
+
+        public TypoCorrection(int maxLevenshteinDistance)
+        {
+            _maxLevenshteinDistance = maxLevenshteinDistance;
```

Comment:
should this check for >= 0 distance?

Reply:
Yes, assuming we keep this as an incoming parameter, that should be checked. Ideally it would be nice to eliminate the fixed value and use some heuristics based on the unmatched token.